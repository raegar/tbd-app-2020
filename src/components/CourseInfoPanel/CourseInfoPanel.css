.courseInfoBox {
	/*margin-top: 3rem !important;*/
	height: 100%;
	background-color: #B2DDDF;
	margin-left: 1rem;
	padding: 1.5em 3em 3em 3em;
}

.titleRow {

}

.courseInfoTitle {
	font-style: bold;
}

.courseTitle {

}

.iframeContainer {
	width: 95%;
}

.uniPageLinkRow {
	
}

.uniPageLink {

}

.icon {
	color: white;
}

.iframeLink {
	width: 100%;
	height: 40em;
	margin: 1em;
	overflow-x: scroll;
}