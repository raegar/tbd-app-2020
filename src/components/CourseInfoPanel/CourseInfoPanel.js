import React, {useState} from 'react';
import "./CourseInfoPanel.css";
import { Row, Container } from "react-bootstrap";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCommentDots } from '@fortawesome/free-solid-svg-icons';
import PropTypes from 'prop-types';
import { SubjectInfo } from '../../global/Constants';
import UCPButton from '../Buttons/UCPButton';
import '../../global/GlobalVariables.js'


function CourseInfoPanel({className, ...props}) {
	const [showEmbed, setShowEmbed] = useState(true);

	return (
		<Container className={"courseInfoBox " + className}>
			<Row className="courseTitleRow">
				{/* Title */}
				<span className="courseInfoTitle"><FontAwesomeIcon className="icon" icon={faCommentDots}/>  Course Info</span>
			</Row>
			<Row className="uniPageLinkRow">
				{/* Uni page link */}
				<UCPButton 
					to="none" 
					onClick={() => setShowEmbed(!showEmbed)} 
					buttonText={(showEmbed ? "Close" : "Open") + " Embed"}
					className="smallbutton"
				/>
				{showEmbed ? <div className="iframeContainer">
					<iframe className="iframeLink" src="https://www.ucp.ac.uk/a-z-courses/" title="Courses at UCP"/>
				</div> : null}
			</Row>
			<Row>
				<a href="https://www.ucp.ac.uk/a-z-courses/">Link to Course information page on UCP website</a>
			</Row>
			{global.SelectedCourse ? <Row className="courseInfoRow">
				<span><b>Course Overview: </b>{SubjectInfo[global.SelectedCourse]}</span>
			</Row> : null}
		</Container>
	);
}

CourseInfoPanel.propTypes = {
	courseSelected: PropTypes.string,
	className: PropTypes.any
};

export default CourseInfoPanel;