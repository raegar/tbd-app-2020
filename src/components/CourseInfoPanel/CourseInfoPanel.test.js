import React from 'react';
import { render, screen, cleanup, wrapper } from '@testing-library/react';
import '@testing-library/jest-dom';
import CourseInfoPanel from './CourseInfoPanel';

afterEach(cleanup);

it("Renders the calculator", () => {
    const { asFragment } = render(<CourseInfoPanel/>);
    expect(asFragment()).toMatchSnapshot();
});