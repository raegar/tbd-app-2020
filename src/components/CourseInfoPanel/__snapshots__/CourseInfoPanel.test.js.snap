// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the calculator 1`] = `
<DocumentFragment>
  <div
    class="courseInfoBox undefined"
  >
    <div
      class="courseTitleRow row"
    >
      <h1
        class="courseInfoTitle"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-comment-dots fa-w-16 icon"
          data-icon="comment-dots"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5c-2.2 2.3-2.8 5.7-1.5 8.7S4.8 480 8 480c66.3 0 116-31.8 140.6-51.4 32.7 12.3 69 19.4 107.4 19.4 141.4 0 256-93.1 256-208S397.4 32 256 32zM128 272c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128 0c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128 0c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z"
            fill="currentColor"
          />
        </svg>
          Course Info
      </h1>
    </div>
    <div
      class="row"
    >
      <button
        class="mediumbutton"
        data-testid="ucpButton"
        type="button"
      >
        Move UCAS Calculator to Here​
      </button>
    </div>
    <div
      class="uniPageLinkRow row"
    >
      <button
        class="smallbutton"
        data-testid="ucpButton"
        type="button"
      >
        Close Embed​
      </button>
      <div
        class="iframeContainer"
      >
        <iframe
          class="iframeLink"
          src="https://www.ucp.ac.uk/a-z-courses/"
          title="Courses at UCP"
        />
      </div>
    </div>
    <div
      class="uniPageLinkRow row"
    >
      <a
        href="https://www.ucp.ac.uk/a-z-courses/"
      >
        Link to Course information page on UCP website
      </a>
    </div>
    <div
      class="ucasCalculatorRow row"
    >
      <p>
        <b>
          UCAS Calculator:
        </b>
      </p>
      <div
        class="ucasCalculator"
      >
        <div>
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <select
                  class="qualInputBox"
                  id="qualifications"
                >
                  <option
                    disabled=""
                  >
                    Please select a qualification...
                  </option>
                  <option
                    class="gradeTypeHeader"
                    disabled=""
                  >
                    A Level and AS
                  </option>
                  <option
                    value="A Level and AS;A Levels"
                  >
                      A Levels
                  </option>
                  <option
                    value="A Level and AS;AS Levels"
                  >
                      AS Levels
                  </option>
                  <option
                    class="gradeTypeHeader"
                    disabled=""
                  >
                    BTEC National (RQF) (From 2016)
                  </option>
                  <option
                    value="BTEC National (RQF) (From 2016);Pearson BTEC Level 3 National Extended Certificate (first teaching from September 2016)"
                  >
                      Pearson BTEC Level 3 National Extended Certificate (first teaching from September 2016)
                  </option>
                </select>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-md-2"
              >
                <div
                  class="mobiletextbox"
                >
                  <label />
                  <input
                    class="subjectInputBox"
                    maxlength="30"
                    minlength="0"
                    placeholder="Subject (optional)"
                    size="20"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="col"
              >
                <br />
              </div>
              <div
                class="col-md-2"
              >
                <br />
                <select
                  class="Dropdown"
                  disabled=""
                >
                  <option
                    value=""
                  />
                  <option
                    value="none"
                  >
                    ...
                  </option>
                </select>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-md-6"
              />
              <div
                class="col"
              >
                <span
                  class="qualPoints"
                  data-testid="qualPoints"
                >
                  Points: 
                  <b>
                    0
                  </b>
                </span>
              </div>
            </div>
            <br />
            <div
              class="row"
            >
              <button
                class="mediumbutton addQualButton"
                data-testid="ucpButton"
                type="button"
              >
                Add another qualification​
              </button>
            </div>
            <span
              class="totalQualPoints"
            >
              Total points: 
              <b
                data-testid="totalPoints"
              >
                0
              </b>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
